File: app/src/main/java/com/meuapttestemobile/data/DataModule.java
package com.meuapttestemobile.data;

import com.meuapttestemobile.App;
import com.meuapttestemobile.data.dataSource.ShotDataSource;
import com.meuapttestemobile.data.repository.ShotRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DataModule {

    @Provides
    @Singleton
    ShotDataSource provideShotRepository(APIService api, App app) {
        return new ShotRepository(api, app);
    }
}

File: app/src/main/java/com/meuapttestemobile/data/DataModule.kt
package com.meuapttestemobile.data

import com.meuapttestemobile.App
import com.meuapttestemobile.data.remote.APIService
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DataModule {

    @Provides
    @Singleton
    fun provideShotRepository(api: APIService, app: App): ShotDataSource = ShotRepository(api, app)
}
